pkg/deploy/elbv2/listener_rule_synthesizer.go

```diff
@@ -3,29 +3,34 @@ package elbv2
 import (
 	"context"
 	awssdk "github.com/aws/aws-sdk-go/aws"
-	elbv2sdk "github.com/aws/aws-sdk-go/service/elbv2"
 	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"sigs.k8s.io/aws-load-balancer-controller/pkg/aws/services"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/deploy/tracking"
 	"sigs.k8s.io/aws-load-balancer-controller/pkg/model/core"
 	elbv2model "sigs.k8s.io/aws-load-balancer-controller/pkg/model/elbv2"
 	"strconv"
 )
 
 // NewListenerRuleSynthesizer constructs new listenerRuleSynthesizer.
-func NewListenerRuleSynthesizer(elbv2Client services.ELBV2, lrManager ListenerRuleManager, logger logr.Logger, stack core.Stack) *listenerRuleSynthesizer {
+func NewListenerRuleSynthesizer(elbv2Client services.ELBV2, trackingProvider tracking.Provider, taggingManager TaggingManager,
+	lrManager ListenerRuleManager, logger logr.Logger, stack core.Stack) *listenerRuleSynthesizer {
 	return &listenerRuleSynthesizer{
-		elbv2Client: elbv2Client,
-		lrManager:   lrManager,
-		logger:      logger,
-		stack:       stack,
+		elbv2Client:      elbv2Client,
+		lrManager:        lrManager,
```

Comment:
seems trackingProvider is not needed here

Reply:
removed